src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -377,6 +478,54 @@ public void OnEventCommand(EventCommandEventArgs command)
                 }
             }
 
+            private void IncrementRefCount(string uniqueIdentifier, EventCommandEventArgs command)
+            {
+                // Could be unsafe if UniqueIdentifier protocol isn't followed
+                if (command.Arguments!.TryGetValue("UniqueIdentifier", out string? uniqueIdentifierArg))
```

Comment:
Should we just reject shared sessions that do not have UniqueIdentifiers?

Reply:
I'm not sure that's possible...the way Shared sessions work, every consumer (e.g. dotnet-monitor) with a SHARED session id will listen to the events, _unless_ the UniqueIdentifier (now renamed to ClientId) has a `MultipleSessionsConfiguredIncorrectlyError`. Without a ClientId, we have no way to communicate to that particular consumer that they shouldn't be allowed to listen to the SHARED session. In theory you could shut down the whole SHARED session, but that doesn't necessarily feel like a great solution either.